Thunder/server/stream_routes.py

```
@@ -1,303 +1,225 @@
+        except (FileNotFound, InvalidHash):
+            work_loads[client_id] -= 1
+            raise
+        except Exception as e:
+            work_loads[client_id] -= 1
+            error_id = secrets.token_hex(6)
+            logger.error(f"Stream error {error_id}: {e}")
+            raise web.HTTPInternalServerError(text="Server error") from e
+        
+    except (InvalidHash, FileNotFound) as e:
+        logger.debug(f"Client error: {type(e).__name__}")
+        raise web.HTTPNotFound(text="Resource not found")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)